src/engine/strat_engine/pool.rs

```diff
@@ -48,12 +59,35 @@ impl StratPool {
         let devices = try!(resolve_devices(paths));
         let bds = try!(initialize(&pool_uuid, devices, MIN_MDA_SECTORS, force));
 
+        let meta_dev;
+        let data_dev;
+        let length;
+        {
+            let mut data_devs = Vec::from_iter(bds.values());
+            let metadata_blockdev = data_devs.remove(0);
```

Comment:
OK. Currently it is guaranteed that bds.len() is at least 2. That's tricky, though. Seems like its worth an assert and another TODO linking it to the check in ```initialize()```.

Also,  maybe tighten up block every so slightly, as:

```
let mut data_devs = Vec::from_iter(bds.values());
meta_dev = try!(LinearDev::new(..., ..., &vec![data_devs.remove(0)]);
data_dev = ...
length = ...
```